src/main/scala-2.11/org/apache/james/gatling/jmap/JmapMailboxes.scala

```diff
@@ -0,0 +1,55 @@
+package org.apache.james.gatling.jmap
+
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+
+import io.gatling.http.request.builder.HttpRequestBuilder
+
+object JmapMailboxes {
```

Comment:
the whole solution is really overkill. I think that you could just design your extractors as checks and then chain calls to these checks after the HttpRequest.